src/compas/geometry/primitives/polyline.py

```diff
@@ -413,3 +413,104 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
         return division_pts
 
     divide_by_length = divide_polyline_by_length
+
+    def extend(self, length):
+        """Extends a polyline by a given length, by modifying the first and/or last point tangentially.
+
+        Parameters:
+        -----------
+        length: float or tuple[float, float]
+            A single length value to extend the polyline only at the end,
+            or two length values to extend at both ends.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Polyline`
+
+        """
+        try:
+            start, end = length
+            self.points[0] = self.points[0] + self.lines[0].vector.unitized().scaled(-start)
+        except TypeError:
+            start = end = length
+        self.points[-1] = self.points[-1] + self.lines[-1].vector.unitized().scaled(end)
+
+    def extended(self, length):
+        """Extends a polyline by a given length, by modifying the first and/or last point tangentially.
+        Returns a copy of the original polyline.
```

Comment:
Only the first line is recognized as the short summary. Therefore this should be something like "Returns a copy of this polyline extended by a given length."